### PR TITLE
Https endpoint

### PR DESCRIPTION
**Pull request by ShimShtein**

### CODE REVIEW

File: app/models/endpoints/https_endpoint.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        yield(connection)
+      end
+    end
+
+    private
+
+    def ca_cert_store
+      store = OpenSSL::X509::Store.new
+      server_ca_certificate&.split(/(?=-----BEGIN)/)&.each do |cert|
+        store.add_cert(OpenSSL::X509::Certificate.new(cert))
```
[the remark this answers is not in the document]
Comment (ShimShtein) [reply app/models/endpoints/https_endpoint.rb]: > If the cert given by the user is not in correct format, the endpoint gets saved and then delivery fails

Already handled in `server_ca_certificate=`

>  Is the cert always required? 

I think so... At least for production use cases you would like to validate that you are indeed talking to the service you wanted. If you want, we can expose the verify_mode, but I would advice against it.

[the remark this answers is not in the document]
Comment (adamruzicka) [reply app/models/endpoints/https_endpoint.rb]: But `server_ca_certificate=` is not used when you set it during initialization `Endpoints::HttpsEndpoint.new(opts.merge(:data => { :server_ca_certificate => 'bogus' }))`

I don't mind always needing the cert, but then we should validate its presence

**Comment by ShimShtein** [comment]
Moved validations into rails validations framework.

**Comment by ShimShtein** [comment]
@adamruzicka should be fixed now.

**Comment by ShimShtein** [comment]
rebased and addressed empty key issue. Now we even have a test for it.

**Comment by adamruzicka** [comment]
Thanks @ShimShtein !

**Pull request merged**